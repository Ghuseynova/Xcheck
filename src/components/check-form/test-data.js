const initialStateEmpty = {
  reviewRequest: {
    id: 'rev-req-9',
    author: 'viktorsipach',
    taskTitle: 'Task 3',
    state: 'published',
    selfGrade: {
      'task-3-bp1': { score: 20, comment: 'Well done!' },
      'task-3-ep1': { score: 15, comment: 'Some things are done, some are not' },
      'task-3-fp1': { score: 0, comment: 'No ticket today' },
    },
    linkToDemo: '#',
    linkToPR: '#',
    crossCheckSessionId: '',
  },
  scopes: [
    {
      id: 1,
      title: 'Basic Scope',
      items: [
        {
          id: 'task-3-bp1',
          title: 'basic_p1',
          category: 'basic',
          score: '10',
          description: 'basic_p1 - You need to do some things',
          mentorCheck: true,
        },
      ],
    },
    {
      id: 2,
      title: 'Extra Scope',
      items: [
        {
          id: 'task-3-ep1',
          title: 'extra_p1',
          category: 'extra',
          score: '30',
          description: 'extra_p1 - You need to do some things',
          mentorCheck: true,
        },
      ],
    },
    {
      id: 3,
      title: 'Fines',
      items: [
        {
          id: 'task-3-fp1',
          title: 'fines_p1',
          category: 'fines',
          score: '10',
          description: 'fines_p1 - You need to do some things',
          mentorCheck: true,
        },
      ],
    },
  ],
};

const initialStateFilled = {
  reviewRequest: {
    id: 'rev-req-9',
    author: 'viktorsipach',
    taskTitle: 'Task 3',
    state: 'published',
    selfGrade: {},
    linkToDemo: '#',
    linkToPR: '#',
    crossCheckSessionId: '',
  },
  scopes: [
    {
      id: 1,
      title: 'Basic Scope',
      items: [
        {
          id: 'task-3-bp1',
          title: 'basic_p1',
          category: 'basic',
          score: '10',
          description: 'basic_p1 - You need to do some things',
          mentorCheck: true,
        },
      ],
    },
    {
      id: 2,
      title: 'Extra Scope',
      items: [
        {
          id: 'task-3-ep1',
          title: 'extra_p1',
          category: 'extra',
          score: '30',
          description: 'extra_p1 - You need to do some things',
          mentorCheck: true,
        },
      ],
    },
    {
      id: 3,
      title: 'Fines',
      items: [
        {
          id: 'task-3-fp1',
          title: 'fines_p1',
          category: 'fines',
          score: '10',
          description: 'fines_p1 - You need to do some things',
          mentorCheck: true,
        },
      ],
    },
  ],
};

export { initialStateEmpty, initialStateFilled };
