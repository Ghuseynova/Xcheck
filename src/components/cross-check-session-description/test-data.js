const initialValues = {
  id: "xcheck-1",
  title: "rss2020Q3react-songbird",
  author: "viktorsipach",
  state: "draft",
  taskTitle: "Task 1",
  crossCheckSessionPeriod: [
    "2020-07-08",
    "2020-07-14"
  ],
  coefficient: 0.2,
  discardMinScore: true,
  discardMaxScore: true,
  minReviewsAmount: 1,
  desiredReviewsAmount: 2,
  attendees: [
    {
      "githubId": "viktorsipach",
      "reviewerOf": [
        "Kipris",
        "Ghuseynova"
      ]
    },
    {
      "githubId": "PavelZaharov1987",
      "reviewerOf": [
        "viktorsipach",
        "Ghuseynova"
      ]
    },
    {
      "githubId": "Ghuseynova",
      "reviewerOf": [
        "Kipris",
        "PavelZaharov1987"
      ]
    },
    {
      "githubId": "Kipris",
      "reviewerOf": [
        "viktorsipach",
        "PavelZaharov1987"
      ]
    }
  ]
}

export default initialValues;
