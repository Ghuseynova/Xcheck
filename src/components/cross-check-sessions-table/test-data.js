const tableData = [
  {
    "id": "xcheck-1",
    "title": "rss2020Q3react-songbird",
    "author": "viktorsipach",
    "state": "draft",
    "taskTitle": "Task 1",
    "crossCheckSessionPeriod": [
      "2020-07-08",
      "2020-07-14"
    ],
    "coefficient": 0.2,
    "discardMinScore": true,
    "discardMaxScore": true,
    "minReviewsAmount": 1,
    "desiredReviewsAmount": 2,
    "attendees": []
  },
  {
    "id": "xcheck-2",
    "title": "rss2020Q3react-xcheck",
    "author": "KatiaR",
    "state": "draft",
    "taskTitle": "Task 2",
    "crossCheckSessionPeriod": [
      "2020-09-15",
      "2020-09-17"
    ],
    "coefficient": 0.7,
    "discardMinScore": true,
    "discardMaxScore": true,
    "minReviewsAmount": 1,
    "desiredReviewsAmount": 2,
    "attendees": []
  }
]

export default tableData;