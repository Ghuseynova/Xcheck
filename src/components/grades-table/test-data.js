const tableData = [
    {
        "task-1-bp1": {
          "score": 15,
          "comment": "Well done!"
        },
        "task-1-ep1": {
          "score": 0,
          "comment": "Some things are done, some are not"
        },
        "task-1-fp1": {
          "score": -20,
          "comment": "No ticket today"
        }
    },
]

export default tableData;