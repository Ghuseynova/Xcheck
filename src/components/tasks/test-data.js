const tasks = [
	{
		"taskScore": 100,
		"title": "Task 1",
		"state": "draft",
		"author": "viktorsipach",
		"description": "Task 1 description",
		"items": [
			{
				"id": "task-1-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-bp2",
				"title": "basic_p2",
				"category": "basic",
				"score": "20",
				"description": "basic_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-ep2",
				"title": "extra_p2",
				"category": "extra",
				"score": "50",
				"description": "extra_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-fp2",
				"title": "fines_p2",
				"category": "fines",
				"score": "10",
				"description": "fines_p2 - You need to do some things",
				"mentorCheck": false
			}
		],
		"taskId": "task-1",
		"id": "task-1"
	},
	{
		"taskScore": 100,
		"title": "Task 2",
		"state": "closed",
		"author": "KatiaR",
		"description": "Task 2 description",
		"items": [
			{
				"id": "task-2-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-2-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-2-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			}
		],
		"taskId": "task-2",
		"id": "task-2"
	},
	{
		"taskScore": 100,
		"title": "Task 3",
		"state": "open",
		"author": "YekaterinaKarakulina",
		"description": "Task 3 description",
		"items": [
			{
				"id": "task-3-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-3-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-3-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			}
		],
		"taskId": "task-3",
		"id": "task-3"
	}]

const columns = [
	{
		title: 'Title',
		dataIndex: 'title',
		key: 'title',
	},
	{
		title: 'Author',
		dataIndex: 'author',
		key: 'author',
	},
	{
		title: 'Description',
		dataIndex: 'description',
		key: 'description',
	},
	{
		title: 'Status',
		key: 'state',
		dataIndex: 'state',
	},
	{
		title: 'Action',
		key: 'taskId',
	},
];

const task =
{
	"taskScore": 100,
	"title": "Task 1",
	"state": "draft",
	"author": "viktorsipach",
	"description": "Task 1 description",
	"items": [
		{
			"id": "task-1-bp11",
			"title": "basic_p1",
			"category": "basic",
			"score": "10",
			"description": "basic_p1 - You need to do some things",
			"mentorCheck": true
		},
		{
			"id": "task-1-bp21",
			"title": "basic_p2",
			"category": "basic",
			"score": "20",
			"description": "basic_p2 - You need to do some things",
			"mentorCheck": false
		},
		{
			"id": "task-1-ep11",
			"title": "extra_p1",
			"category": "extra",
			"score": "30",
			"description": "extra_p1 - You need to do some things",
			"mentorCheck": true
		},
		{
			"id": "task-1-ep21",
			"title": "extra_p2",
			"category": "extra",
			"score": "50",
			"description": "extra_p2 - You need to do some things",
			"mentorCheck": false
		},
		{
			"id": "task-1-fp11",
			"title": "fines_p1",
			"category": "fines",
			"score": "10",
			"description": "fines_p1 - You need to do some things",
			"mentorCheck": true
		},
		{
			"id": "task-1-fp21",
			"title": "fines_p2",
			"category": "fines",
			"score": "10",
			"description": "fines_p2 - You need to do some things",
			"mentorCheck": false
		}
	],
	"taskId": "task-11",
	"id": "task-11"
};

export { tasks, columns, task };