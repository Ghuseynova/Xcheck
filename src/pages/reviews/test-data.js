const reviewsData = [
    {
        "id": "rev-id-1",
        "requestId": "rev-req-1",
        "author": "KatiaR",
        "requestor": "viktorsipach",
        "taskTitle": "Task 1",
        "state": "published",
        "comments": [
          {
            "author": "viktorsipach",
            "avatar": "https://avatars1.githubusercontent.com/u/54073656?v=4",
            "content": "Test saga!"
          },
          {
            "author": "viktorsipach",
            "avatar": "https://avatars1.githubusercontent.com/u/54073656?v=4",
            "content": "Yes!"
          },
          {
            "author": "viktorsipach",
            "avatar": "https://avatars1.githubusercontent.com/u/54073656?v=4",
            "content": "Well done!"
          },
          {
            "author": "viktorsipach",
            "avatar": "https://avatars1.githubusercontent.com/u/54073656?v=4",
            "content": "Good!"
          }
        ],
        "grade": {
          "items": {
            "task-1-bp1": {
              "score": 15,
              "comment": "Well done!"
            },
            "task-1-ep1": {
              "score": 0,
              "comment": "Some things are done, some are not"
            },
            "task-1-fp1": {
              "score": -20,
              "comment": "No ticket today"
            }
          },
          "total": "50"
        }
    }
]

export default reviewsData;