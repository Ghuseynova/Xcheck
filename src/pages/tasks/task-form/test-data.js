const taskFormValues = {
  author: "KatiaR",
  description: "Some task description",
  items: [
    {
      category: "basic",
      description: "",
      id: "53fa08ba-2c49-46eb-bd62-a1a793d84f75",
      mentorCheck: true,
      score: 20,
      title: "basic",
    },
    {
      category: "extra",
      description: "extra item description",
      id: "7bda6a67-e531-406c-8c44-4a8c7c43398c",
      mentorCheck: false,
      score: 50,
      title: "extra"
    },
    {
      category: "fines",
      description: "",
      id: "278222de-235e-4bef-9c3a-3935ebe08323",
      mentorCheck: true,
      score: 5,
      title: "fine",
    }
  ],
  state: "open",
  title: "some task"
}

export default taskFormValues;
