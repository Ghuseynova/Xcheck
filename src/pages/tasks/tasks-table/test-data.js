const tasks = [
	{
		"taskScore": 100,
		"title": "Task 1",
		"state": "draft",
		"author": "viktorsipach",
		"description": "Task 1 description",
		"items": [
			{
				"id": "task-1-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-bp2",
				"title": "basic_p2",
				"category": "basic",
				"score": "20",
				"description": "basic_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-ep2",
				"title": "extra_p2",
				"category": "extra",
				"score": "50",
				"description": "extra_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-fp2",
				"title": "fines_p2",
				"category": "fines",
				"score": "10",
				"description": "fines_p2 - You need to do some things",
				"mentorCheck": false
			}
		],
		"taskId": "task-1",
		"id": "task-1"
	},
	{
		"taskScore": 100,
		"title": "Task 2",
		"state": "closed",
		"author": "KatiaR",
		"description": "Task 2 description",
		"items": [
			{
				"id": "task-2-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-2-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-2-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			}
		],
		"taskId": "task-2",
		"id": "task-2"
	}
]
export default tasks;