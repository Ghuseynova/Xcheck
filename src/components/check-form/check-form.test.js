import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import CheckForm from './check-form';
import { initialStateEmpty, initialStateFilled } from './test-data';
import { mount } from '../../../jest.setup';

const mockStore = configureStore();
const mockDispatchFn = jest.fn();

describe('Render CheckForm When SelfGrade Empty', () => {
  it('render correctly', () => {
    const wrapper = mount(
      <Provider store={mockStore()}>
        <MemoryRouter>
          <CheckForm
            {...initialStateEmpty}
            dispatch={mockDispatchFn}
            detailIds={{}}
            commentFieldIds={{}}
            commentIds={{}}
            changedInputIds={{}}
          />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Render CheckForm When SelfGrade Fillled', () => {
  it('render correctly', () => {
    const wrapper = mount(
      <Provider store={mockStore()}>
        <MemoryRouter>
          <CheckForm
            {...initialStateFilled}
            dispatch={mockDispatchFn}
            detailIds={{}}
            commentFieldIds={{}}
            commentIds={{}}
            changedInputIds={{}}
          />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
