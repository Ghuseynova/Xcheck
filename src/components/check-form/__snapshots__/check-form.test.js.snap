// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CheckForm When SelfGrade Empty render correctly 1`] = `ReactWrapper {}`;

exports[`Render CheckForm When SelfGrade Fillled render correctly 1`] = `ReactWrapper {}`;
